### PR TITLE
 [MGX-632] 3rdparty rewards #593 

### PR DESCRIPTION
**Pull request by mateuszaaa**
This PR introduces 3rdparty rewards feature.

Deprecated RPC calls:

    rpc XYK::claim_rewards_all -> ProofOfStake::claim_native_rewards

Deprecated Extirnsics:

    ProofOfStake::activate_liquidity -> ProofOfStake::activate_liquidity_for_native_rewards
    ProofOfStake::deactivate_liquidity -> ProofOfStake::deactivate_liquidity_for_native_rewards
    ProofOfStake::claim_rewards_all -> ProofOfStake::claim_native_rewards

### CODE REVIEW

File: runtime/mangata-kusama/Cargo.toml [hunk 1]
```diff
@@ -142,6 +150,7 @@ std = [
 	"orml-xtokens/std",
 	"pallet-aura/std",
 	"pallet-authorship/std",
+	"pallet-balances/std",
```
Comment (vanderian) [inline runtime/mangata-kusama/Cargo.toml:153]: what are we using `pallet-balances` for

**Review by vanderian — APPROVED** [review]

File: pallets/proof-of-stake/src/lib.rs [hunk 2]
```diff
@@ -226,126 +659,582 @@ pub mod pallet {
...
-				sender,
-				liquidity_token_id,
-				amount,
-				use_balance_from,
-			)
+			<Self as ProofOfStakeRewardsApi<
+				AccountIdOf<T>,
+				BalanceOf<T>,
+				CurrencyIdOf<T>,
+			>>::activate_liquidity(sender, liquidity_token_id, amount, use_balance_from)
```
[the remark this answers is not in the document]
Comment (mateuszaaa) [reply pallets/proof-of-stake/src/lib.rs]: yep!

File: pallets/proof-of-stake/src/reward_info.rs [hunk 3]
```diff
@@ -170,12 +212,187 @@ where
...
+	+ Into<u128>
+	+ AtLeast32BitUnsigned
+	+ FullCodec
+	+ Copy
+	+ Default
+	+ sp_std::fmt::Debug
+	+ scale_info::TypeInfo
+	+ MaxEncodedLen
+{
+}
```
[the remark this answers is not in the document]
Comment (mateuszaaa) [reply pallets/proof-of-stake/src/reward_info.rs]: its used now :)

File: runtime/mangata-kusama/Cargo.toml (hunk 1 above)
Comment (mateuszaaa) [reply runtime/mangata-kusama/Cargo.toml]: removedd

File: pallets/parachain-staking/src/benchmarks.rs [hunk 4]
```diff
@@ -1685,257 +1685,257 @@ benchmarks! {
...
-			}
-		}
-
-	}: {<pallet_session::Pallet::<T>  as frame_support::traits::Hooks<_>>::on_initialize(<frame_system::Pallet<T>>::block_number());}
-	verify {
-		assert_eq!(pallet_session::Pallet::<T>::current_index() as u32, 7u32);
-		assert_eq!(Pallet::<T>::round().current as u32, 7u32);
-		assert_eq!(w as usize, candidates.iter().filter_map(|c| RoundCollatorRewardInfo::<T>::get(c.clone(), 5u32)).count());
-	}
+	// active_session_change {
```
Comment (majusko) [inline pallets/parachain-staking/src/benchmarks.rs:1688]: Why did we comment out this section?

**Review by majusko — CHANGES_REQUESTED** [review]

File: pallets/proof-of-stake/rpc/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+	generic::BlockId,
+	traits::{Block as BlockT, MaybeDisplay, MaybeFromStr},
+};
+use sp_std::convert::{TryFrom, TryInto};
+use std::sync::Arc;
+
+#[rpc(client, server)]
+pub trait ProofOfStakeApi<BlockHash, Balance, TokenId, AccountId> {
+	#[method(name = "pos_calculate_native_rewards_amount")]
+	fn calculate_native_rewards_amount(
```
Comment (majusko) [inline pallets/proof-of-stake/rpc/src/lib.rs:25]: Collect all breaking changes into the Jira ticket pls

File: pallets/proof-of-stake/rpc/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+use sp_core::U256;
+use sp_rpc::number::NumberOrHex;
+use sp_runtime::{
+	generic::BlockId,
+	traits::{Block as BlockT, MaybeDisplay, MaybeFromStr},
+};
+use sp_std::convert::{TryFrom, TryInto};
+use std::sync::Arc;
+
+#[rpc(client, server)]
```
Comment (majusko) [inline pallets/proof-of-stake/rpc/src/lib.rs:22]: If possible, please add documentation to API, describing what it is doing and what is the expected result.

**Review by majusko — CHANGES_REQUESTED** [review]

File: pallets/parachain-staking/src/benchmarks.rs (hunk 4 above)
Comment (mateuszaaa) [reply pallets/parachain-staking/src/benchmarks.rs]: just for testing, reverted

**Review by vanderian — APPROVED** [review]

**Review by majusko — APPROVED** [review]

**Pull request merged**